test/boost/sstable_compaction_test.cc

```diff
@@ -299,30 +299,29 @@ static std::vector<sstables::shared_sstable> get_candidates_for_leveled_strategy
 static future<std::vector<sstables::generation_type::int_t>> compact_sstables(test_env& env, std::vector<sstables::generation_type::int_t> generations_to_compact,
         sstables::generation_type::int_t new_generation, bool create_sstables, uint64_t min_sstable_size, compaction_strategy_type strategy) {
     BOOST_REQUIRE(smp::count == 1);
+    return seastar::async(
```

Comment:
A suggestion. There are several patches that toss seastar::async() stuff around sstable tests. I might make sense to submit them altogether as a separate PR

Reply:
All the patches that clean up / refactor tests can be upstreamed quickly.